### PR TITLE
feat(planner/elixir): Add Elixir support

### LINKED ISSUE #44: elixir support
We are looking for a new contributor to help us add elixir support in zbpack.

Zeabur doesn's support [elixir](https://elixir-lang.org/) services now, but as a great programming language, some users may want to deploy elixir services on Zeabur.

If you have any questions or need help getting started, feel free to reach out to us in our [Discord server.](https://discord.gg/DrdGCvXEyY) We are happy to provide guidance and support.

### PR DESCRIPTION
**Pull request by Charlie-117**
* supports Phoenix, Ecto apps

closes #44

### CODE REVIEW

File: internal/elixir/identify.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+func (i *identify) PlanMeta(options plan.NewPlannerOptions) types.PlanMeta {
+	ElixirVer, err := DetermineElixirVersion(options.Source)
+	if err != nil {
+		panic(err)
+	}
+
+	ElixirEcto, err := CheckElixirEcto(options.Source)
+	if err != nil {
+		panic(err)
+	}
```
Comment (pan93412) [inline internal/elixir/identify.go:36]: Better to not `panic()` when something wrong. Seems like we need to re-design the API? cc @aFlyBird0

File: internal/elixir/plan.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+			case "1.13":
+				return "1.13", nil
+			case "1.14":
+				return "1.14", nil
+			case "1.15":
+				return "1.15", nil
+			// For future versions use latest
+			default:
+				return "latest", nil
+			}
```
Comment (pan93412) [inline internal/elixir/plan.go:48]: Better to move this map to another file under this module.

File: internal/elixir/templates/template.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# Set working directory
+WORKDIR /app
+
+# Set environment variables
+ENV MIX_ENV=prod
+ENV PORT=8080
+
+# Copy the project files to the working directory
+COPY . /app
```
Comment (pan93412) [inline internal/elixir/templates/template.Dockerfile:12]: Better to not `COPY . /app` so soon – it invalidates the layer cache. https://dev.to/shmink/multi-stage-docker-and-elixir-releases-1ce2

**Review by pan93412 — CHANGES_REQUESTED** [review]

**Comment by itsalicelee** [comment]
@Charlie-117 Thanks for your contribution!
Just try out [the official phoenix template](https://github.com/phoenixframework/phoenix_template), it worked but didn't show the framework information.
image
image1

File: internal/elixir/templates/template.Dockerfile (hunk 3 above)
Comment (Charlie-117) [reply internal/elixir/templates/template.Dockerfile]: Done.

**Comment by Charlie-117** [comment]
> @Charlie-117 Thanks for your contribution! Just try out [the official phoenix template](https://github.com/phoenixframework/phoenix_template), it worked but didn't show the framework information. image image1

Earlier the commit only supported Phoenix framework so I didn't think of showing the framework's name but now I have tried to add support for regular Elixir apps along with Phoenix framework.

It now displays the framework name if present (Phoenix only for now).

**Review by pan93412 — APPROVED** [review]
Thanks for your contribution!

**Comment by Charlie-117** [comment]
@MichaelYuhe Can you please add `gssoc23` and `level2` labels ? Thanks.

**Comment by MichaelYuhe** [comment]
Thanks for your contribution!

**Pull request merged**